evm_loader/program/src/precompile_contracts.rs

```diff
@@ -382,13 +399,16 @@ pub fn query_account<'a, B: AccountStorage>(
             let offset = U256::from_big_endian_fast(offset).as_usize();
             let length = U256::from_big_endian_fast(length).as_usize();
             debug_print!("query_account get data {} {} {}", account_address, offset, length);
-            let data = state.query_solana_account_data(account_address, offset, length);
-            if let Some(data) = data {
-                debug_print!("query_account data result: {:?}", data);
-                return Capture::Exit((ExitReason::Succeed(evm::ExitSucceed::Returned), data));
+            match state.query_solana_account().data(&account_address, offset, length) {
+                Ok(data) => {
+                    debug_print!("query_account data result: {:?}", data);
+                    Capture::Exit((ExitReason::Succeed(evm::ExitSucceed::Returned), data))
+                },
+                Err(err) => {
+                    let revert_message = format!("QueryAccount.data failed: {:?}", err).as_bytes().to_vec();
```

Comment:
Please use `{}` as formatter. And `enum Error` should derive `Display`

Reply:
Done.